pkg/integrations/prometheus/promql/metricsSearchHandler.go

```diff
@@ -384,6 +387,112 @@ func ProcessGetAllMetricTagsRequest(ctx *fasthttp.RequestCtx, myid uint64) {
 	ctx.SetStatusCode(fasthttp.StatusOK)
 }
 
+func ProcessGetMetricTimeSeriesRequest(ctx *fasthttp.RequestCtx, myid uint64) {
+	rawJSON := ctx.PostBody()
+	if len(rawJSON) == 0 {
+		utils.SendError(ctx, "empty json body received", "", nil)
+		return
+	}
+
+	readJSON := make(map[string]interface{})
+	var jsonc = jsoniter.ConfigCompatibleWithStandardLibrary
+	decoder := jsonc.NewDecoder(bytes.NewReader(rawJSON))
+	err := decoder.Decode(&readJSON)
+	if err != nil {
+		utils.SendError(ctx, "Failed to parse request body", "", err)
+		return
+	}
+
+	start, ok := readJSON["start"].(float64)
+	if !ok {
+		utils.SendError(ctx, "Failed to parse 'start' from request body", "", nil)
+		return
+	}
+
+	end, ok := readJSON["end"].(float64)
+	if !ok {
+		utils.SendError(ctx, "Failed to parse 'end' from request body", "", nil)
+		return
+	}
+
+	queryInterfaces, ok := readJSON["queries"].([]interface{})
+	if !ok {
+		utils.SendError(ctx, "Failed to parse 'queries' from JSON body", "", nil)
+		return
+	}
+
+	queries := make([]map[string]interface{}, len(queryInterfaces))
+	for i, qi := range queryInterfaces {
+		queryMap, ok := qi.(map[string]interface{})
+		if !ok {
+			utils.SendError(ctx, "Failed to parse 'query' from JSON body", "", nil)
+			return
+		}
+
+		_, nameOk := queryMap["name"].(string)
+		_, queryOk := queryMap["query"].(string)
```

Comment:
a better way to do this is check for one thing, if it is not good then return error, else move on to next check.
also use `ok` for the same call, it reduces stack memory